builders/build/substrate-api/py-substrate-interface.md

```diff
@@ -0,0 +1,311 @@
+
+In this section, you will learn how to query for on-chain information of Moonbeam networks using Python Substrate Interface library. 
+
+### Accessing Runtime Constants {: #accessing-runtime-constants }
+
+All runtime constants are provided in the metadata. You can use the [`get_metadata_constants`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.SubstrateInterface.get_metadata_constants){target=_blank} method to see a list of available runtime constants within Moonbeam network's metadata. 
+
+Runtime constants available in the metadata can be queried through the [`get_constant`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.SubstrateInterface.get_constant){target=_blank} method.
+
+```python
+# Import
+from substrateinterface import SubstrateInterface
+
+# Construct the API provider
+ws_provider = SubstrateInterface(
+    url="{{ networks.moonbase.wss_url }}",
+)   
+
+# List of available runtime constants in the metadata
+constant_list = ws_provider.get_metadata_constants()
+print(constant_list)
+
+# Retrieve the Existential Deposit constant on Moonbeam, which is 0. 
+constant = ws_provider.get_constant("Balances", "ExistentialDeposit")
+print(constant.value)
+
+```
+
+### Retrieve Blocks and Extrinsics {: #retrieve-blocks-and-extrinsics }
+
+You can retrieve basic information about Moonbeam networks, such as blocks and extrinsics, using the Python Substrate Interface API. 
+
+To retrieve a block, you can use the [`get_block`](https://polkascan.github.io/py-substrate-interface/#substrateinterface.SubstrateInterface.get_block){target=_blank} method. You can also access extrinsics and their data fields inside a block object, which is simply a Python dictionary. 
+
+```python
+# Import
+from substrateinterface import SubstrateInterface
+
+# Construct the API provider
+ws_provider = SubstrateInterface(
+    url="{{ networks.moonbase.wss_url }}",
+)   
+
+# Retrieve the latest block
+block = ws_provider.get_block()
+
+# Retrieve the latest finalized block
+block = ws_provider.get_block_header(finalized_only = True)
+
+# Retrieve a block given its hash
+block_hash = "0xa499d4ebccdabe31218d232460c0f8b91bd08f72aca25f9b25b04b6dfb7a2acb"
+block = ws_provider.get_block(block_hash=block_hash)
+
+for extrinsic in block['extrinsics']:
+
+    if 'address' in extrinsic:
+        signed_by_address = extrinsic['address'].value
+    else:
+        signed_by_address = None
+
+    print('\nPallet: {}\nCall: {}\nSigned by: {}'.format(
+        extrinsic["call"]["call_module"].name,
+        extrinsic["call"]["call_function"].name,
+        signed_by_address
+    ))
+
+```
+
+!!! note
+    The block hash used in the above code sample is the Substrate block hash. The standard methods in Python Substrate Interface assume you are using the Substrate version of primitives, such as block or tx hashes.
+
+
+#### Subscribe to New Block Headers {: #subscribe-to-new-block-headers }
```

Comment:
I try to avoid using the #### headers... can it just be a ### or what do you think?